src/RestAPI.cpp

```diff
@@ -18,57 +18,29 @@ using namespace httpserver;
 
 static string const CONFIG_SECTION = "RestAPI";
 
-static TResult<int> parsePort(string const &portStr) {
-  int port = -1;
-  size_t idx;
-
-  // check if the configuration value for "port" is an integer
-  try {
-    port = std::stoi(portStr, &idx);
-  } catch (invalid_argument const &) {
-    return Error(ErrorCode::InvalidFormat,
-                 "RestAPI.handleRequests: Configuration value '" +
-                     CONFIG_SECTION + ".port' is not an integer");
-  }
-
-  // check if the whole string is an integer (idx is set to one past the end)
-  if (idx != portStr.size()) {
-    return Error(ErrorCode::InvalidFormat,
-                 "RestAPI.handleRequests: Configuration value for '" +
-                     CONFIG_SECTION + ".port' must only consist of digits");
-  }
-
-  if (port < 0 || port > 65535) {
-    return Error(ErrorCode::InvalidValue, "Port value is out of range");
-  }
-
-  return port;
-}
-
 TResultOpt RestAPI::handleRequests() {
   auto configHandler = ConfigHandler::getInstance();
 
-  TResult<string> configPort = configHandler->getValue(CONFIG_SECTION, "port");
+  TResult<int> configPort = configHandler->getValueInt(CONFIG_SECTION, "port");
   if (holds_alternative<Error>(configPort)) {
-    return std::get<Error>(configPort);
+    return get<Error>(configPort);
   }
 
-  auto parsedPort = parsePort(std::get<string>(configPort));
-  if (holds_alternative<Error>(parsedPort)) {
-    return std::get<Error>(parsedPort);
-  }
+  int port = get<int>(configPort);
 
-  int port = std::get<int>(parsedPort);
+  if (port < 0 || port > 65535) {
+    return Error(ErrorCode::InvalidValue,
+                 "RestAPI.handleRequests: Port value is out of range");
+  }
 
   auto webserverParams =
-      create_webserver(port)                                         //
-          .not_found_resource(RestEndpointHandler::NotFoundHandler)  //
-          .internal_error_resource(
-              RestEndpointHandler::InternalErrorHandler)  //
-          .no_regex_checking()                            //
-          .single_resource()                              //
-          .no_basic_auth()                                //
-          .no_digest_auth()                               //
+      create_webserver(port)
+          .not_found_resource(RestEndpointHandler::NotFoundHandler)
+          .internal_error_resource(RestEndpointHandler::InternalErrorHandler)
```

Comment:
The empty comments were there to prevent clang-format to inline the chaining calls (and make them unreadable)

Reply:
to me, this still looks exactly the same, except this one removed (unnecessary) newline..?

Reply:
maybe the behaviour changed as i added more methods. if it is ok now then i am fine